### PR TITLE
RFC: Clarify stable features usage

### PR DESCRIPTION
**Pull request by JeromeJu**
# Changes
This commit helps clarify the usage of stable features to be apparent in also apis other than GA.

/kind documentation
part of: https://github.com/tektoncd/pipeline/issues/6965

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [n/a] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [n/a] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [n/a] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [n/a] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: docs/additional-configs.md [hunk 1]
```diff
@@ -256,7 +256,7 @@ The default is `false`. For more information, see the [associated issue](https:/
 
 - `enable-api-fields`: When using v1beta1 APIs, setting this field to "stable" or "beta"
 enables [beta features](#beta-features). When using v1 APIs, setting this field to "stable"
-allows only stable features, and setting it to "beta" allows only beta features.
```
Comment (khrm) [inline docs/additional-configs.md:259]: Isn't the behavior the same when we use v1beta1 API?  
` When using v1 APIs, setting this field to "stable" allows only stable features,`

Comment (JeromeJu) [reply docs/additional-configs.md:259]: Thanks for the question! 
I am afraid that it isn't the case at the moment. For examples, 'stable' feature behaviors like resolver pass validations for `stable` `enable-api-fields` in v1beta1 but it does not in v1.

**Review by Yongxuanzhang — COMMENTED** [review]
/lgtm

**Comment by dibyom** [comment]
@JeromeJu one question I have is what does it mean for a feature to be GA when the API version is alpha 🤔

**Comment by JeromeJu** [comment]
> @JeromeJu one question I have is what does it mean for a feature to be GA when the API version is alpha 🤔

Good point as I think this is also part of the issue TEP0138 is trying to address. I think we should find a way to elaborate that feature stability(versioning) shall be independent from API version. But not too sure if where this PR modifies is the place for such clarifications?

**Comment by lbernick** [comment]
What's a bit confusing about this is right now we don't really have a notion of "stable features"-- it's just everything that's not an alpha or beta feature. I think this is why the question keeps coming up of why you'd describe fields of an alpha CRD as "stable". The way Kubernetes describes this is IMO a bit more clear: they talk about the feature lifecycle and explain that features "graduate" to GA after being beta, so a "stable" feature is a feature that has "graduated" from beta and no longer needs a feature gate. Similarly for us, a stable feature is something that was previously enabled by "enable-api-fields", but no longer is (although this isn't how I'd define a "stable" feature in user-facing docs).  Maybe adopting some of Kubernetes' language around feature lifecycle could help clarify this confusion, especially if we move to per-feature flags. (Really it's just a different way of describing the same thing.)

**Comment by JeromeJu** [comment]
> Maybe adopting some of Kubernetes' language around feature lifecycle could help clarify this confusion, especially if we move to per-feature flags. (Really it's just a different way of describing the same thing.)

That makes sense. I will open up another PR for this as introducing features lifecycle sounds like a somewhat related but different topic than api_compatibility_policy itself. I am not sure if a separate doc is a better place?

**Comment by lbernick** [comment]
A separate PR is totally fine; I think this doc is an appropriate place to describe feature lifecycle since it's very related to feature stability level. It might also be less confusing to wait until TEP 138 is resolved, especially if we do decide to adopt per-feature flags. Up to you!

**Comment by JeromeJu** [comment]
> A separate PR is totally fine; I think this doc is an appropriate place to describe feature lifecycle since it's very related to feature stability level. It might also be less confusing to wait until TEP 138 is resolved, especially if we do decide to adopt per-feature flags. Up to you!

I agree that this could be one of the changes under TEP0138:) Let's wait till it gets merged.

**Comment by jerop** [comment]
/hold 

https://github.com/tektoncd/community/pull/1034 under discussion

**Comment by JeromeJu** [comment]
> @JeromeJu @jerop what is the status of this one ?

Thanks for the review, this has been made in https://github.com/tektoncd/pipeline/pull/6941 PTAL! . Suggesting that we could close this:)

**Comment by vdemeester** [comment]
Alright, let's close it then.

**Pull request closed without merging**